components/textbook-demo/ExternalRecommendedReadings.vue

```diff
@@ -5,22 +5,24 @@
     </h2>
     <AppLink
       v-for="link in links"
-      :key="link"
+      :key="link.label"
       class="external-recommended-readings__link"
-      :url="link"
+      :segment="link.segment"
+      :url="link.url"
     >
-      {{ link }}
+      {{ link.label }}
     </AppLink>
   </section>
 </template>
 
 <script lang="ts">
 import Vue from 'vue'
 import { Component, Prop } from 'vue-property-decorator'
+import { SegmentData } from '~/constants/appLinks'
 
 @Component
 export default class ExternalRecommendedReadings extends Vue {
-  @Prop({ type: Array, required: true }) links!: string[]
+  @Prop({ type: Array, required: true }) links!: SegmentData[]
```

Comment:
Should be a GeneralLink, not a SegmentData type

Reply:
Good catch! Thanks a lot 🙂